src/runtime/serialization/serialization.cpp

```diff
@@ -0,0 +1,115 @@
+#include "hipSYCL/runtime/serialization/serialization.hpp"
+#include "hipSYCL/runtime/backend.hpp"
+#include "hipSYCL/runtime/dag.hpp"
+#include "hipSYCL/runtime/device_id.hpp"
+#include "hipSYCL/runtime/operations.hpp"
+#include "hipSYCL/sycl.hpp"
+#include "hipSYCL/sycl/access.hpp"
+#include "hipSYCL/sycl/item.hpp"
+#include "hipSYCL/sycl/range.hpp"
+#include "hipSYCL/sycl/serialization/serialization.hpp"
```

Comment:
Why do we need to include the serialization header from the SYCL interface?

Reply:
Since we are using the `<<` operator for `sycl::access::mode` and `sycl::access::target` in the `buffer_memory_requirement::dump` function on line 59